src/colour.c

```diff
@@ -35,37 +35,17 @@ t_colour	ft_add_colour(t_colour c1, t_colour c2)
 
 t_colour	ft_hadamard_colour(t_colour c1, t_colour c2)
 {
+	t_interval	interval;
+
+	interval.min = 0.0;
+	interval.max = 1.0;
 	return ((t_colour){
-		.r = c1.r * c2.r,
-		.b = c1.b * c2.b,
-		.g = c1.g * c2.g
+		.r = ft_clamp(c1.r * c2.r, interval),
+		.b = ft_clamp(c1.b * c2.b, interval),
+		.g = ft_clamp(c1.g * c2.g, interval)
```

Comment:
It's only necessary in ft_hadamard_colour as it is the last operation in the lighting calculation. 

On the contrary it would falsify the result if applied to ft_add_colour as information would get lost in the middle of the process. So it should only happen at this point.